src/Book_Appointment.py

```diff
@@ -44,57 +111,112 @@ def getAppointment ():
         timeframe.rowconfigure(6, weight=1)
         timeframe.rowconfigure(7, weight=2)
 
-        submit = ttk.Button(timeframe, text='Submit', command=lambda:submission(var)) #, onClick=sub(date,time)
+        submit = ttk.Button(timeframe, text='Submit', command=lambda:submission(var,date_var)) #, onClick=sub(date,time)
         submit.grid(row=7, column=0)
         #need to work on the message dialog box
 
-    def submission(time):
-        doc=clicked.get()
-        pur=purpose_var.get()
-        t=time.get()
-        times=["10.30am", "11.00am","11.30am","12.00pm","12.30pm"]
-        var1= "Date: "+date_selected +"\nTime: "+times[t]+"\nDoctor: "+doc+"\nPurpose: "+pur
-        msg=tk.messagebox.askquestion("Are you sure?", var1 );
-        # if(msg=="yes"):
-        #sub()
-        #connect to database
 
-            
+    def submission(time,date):
+        if(purpose_var.get()!=""):
+            doc=clicked.get()
+            pur=purpose_var.get()
+            t=time.get()
+            date_selected=date
+            times=["10:30am", "11:00am","11:30am","12:00pm","12:30pm"]
+            var1= "Date: "+date_selected +"\nTime: "+times[t]+"\nDoctor: "+doc+"\nPurpose: "+pur
+            msg=tk.messagebox.askquestion("Are you sure?", var1 );
+            # if(msg=="yes"):
```

Comment:
Mugs idhar yes or no ka if- else nahi hai so probably that's why when you clicked yes and no- db mein commit ho raha tha- ek baar ye check karle.